### PR TITLE
♻️ Fix pandas indexing warnings

### LINKED ISSUE #663: Fix pandas indexing warnings
The tests give us several warnings like this:

```
tests/ingest_runs/test_genomic_data_loader.py::test_load_harmonized_genomic_files
tests/ingest_runs/test_genomic_data_loader.py::test_load_specimen_harmonized_gf_links
tests/ingest_runs/test_genomic_data_loader.py::test_load_seq_exp_harmonized_genomic_files
  /Users/santiag/Projects/kf-api-study-creator/creator/ingest_runs/genomic_data_loader.py:204: SettingWithCopyWarning:
  A value is trying to be set on a copy of a slice from a DataFrame.
  Try using .loc[row_indexer,col_indexer] = value instead

  See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
    df[CONCEPT.GENOMIC_FILE.HARMONIZED] = True
```

We should set the column values in the proper way

### PR DESCRIPTION
**Pull request by gsantia**
Closes #663 

When running tests we get a bunch of `SettingWithCopyWarning` notices from pandas.
`GenomicDataLoader` is refactored to get rid of these and avoid modifying copies of DataFrames.

### CODE REVIEW

File: creator/ingest_runs/genomic_data_loader.py [hunk 1]
```diff
@@ -115,10 +115,14 @@ def load_specimen_harmonized_gf_links(self, genomic_df):
                 CONCEPT.GENOMIC_FILE.TARGET_SERVICE_ID,
             ]
         ]
-        biospec_gen_df[CONCEPT.BIOSPECIMEN.ID] = biospec_gen_df[
-            CONCEPT.BIOSPECIMEN.TARGET_SERVICE_ID
-        ]
-        biospec_gen_df[CONCEPT.BIOSPECIMEN_GENOMIC_FILE.VISIBLE] = True
+        biospec_gen_df = biospec_gen_df.assign(
```
Comment (dankolbman) [inline creator/ingest_runs/genomic_data_loader.py:118]: If I'm reading this correctly, you are simply renaming this column? Could you use `rename` instead?

Comment (gsantia) [reply creator/ingest_runs/genomic_data_loader.py:118]: No, I'm creating a new `CONCEPT.BIOSPECIMEN.ID` column with the same values as the `CONCEPT.BIOSPECIMEN.TARGET_SERVICE_ID` column. The ingest library requires both

Comment (dankolbman) [reply creator/ingest_runs/genomic_data_loader.py:118]: I see, in that case, does `df.loc[:, CONCEPT.BIOSPECIMEN.ID] = df['CONCEPT.BIOSPECIMEN.TARGET_SERVICE_ID]` work? It seems a lot less convoluted.

File: creator/ingest_runs/genomic_data_loader.py [hunk 2]
```diff
@@ -199,10 +205,15 @@ def _prep_harmonized_gf_df(self, df):
...
             "Source Read": CONCEPT.GENOMIC_FILE.SOURCE_FILE,
         }
-        df = df[list(col_rename.keys())]
-        df.rename(columns=col_rename, inplace=True)
-        df[CONCEPT.GENOMIC_FILE.HARMONIZED] = True
-        df[CONCEPT.GENOMIC_FILE.VISIBLE] = True
+
+        df = df[list(col_rename.keys())].rename(columns=col_rename)
+
+        df = df.assign(
```
Comment (dankolbman) [inline creator/ingest_runs/genomic_data_loader.py:211]: Does using `df.loc` work here?

File: creator/ingest_runs/genomic_data_loader.py (hunk 1 above)
Comment (gsantia) [reply creator/ingest_runs/genomic_data_loader.py:118]: Yeah you're right

Comment (gsantia) [reply creator/ingest_runs/genomic_data_loader.py:118]: That still gives the same warning

**Review by dankolbman — APPROVED** [review]
# :bear:

**Pull request merged**